synapse/tests/test_lib_time.py

```diff
@@ -67,10 +70,11 @@ def test_time_parse_tz(self):
         self.eq(s_time.parse('20200707162953+00:00'), 1594139393000)
         self.eq(s_time.parse('20200707162953-04:00'), 1594153793000)
 
-        # A malformed timezone can still be parsed incorrectly
+        # A malformed timezone can still be parsed
+        self.eq(s_time.parse('20200707162953+423'), 1594123613000)
```

Comment:
This would result in a change of norm behavior

Reply:
Could we tweak to limit the totally undecorated cases to requiring the : ?